### PR TITLE
[fix] Get correct locale with country from browser

### LINKED ISSUE #2525: Display wrong language in the search interface.
**Version of Searx, commit number if you are using on master branch and stipulate if you forked Searx**

**How did you install Searx?**

Just use it in the web browser.
**What happened?**

The UI language is wrong.
**How To Reproduce**

Open searx search page.
**Expected behavior**

See the right language.
**Screenshots & Logs**

圖片

**Additional context**

I'm using Traditional Chinese, but it displays the Simplified Chinese.
https://www.transifex.com/asciimoo/searx/dashboard/
It is fully translated to my language, but it still displays Simplified Chinese.
It also searches lots of websites from China, although I have selected Taiwan.

### PR DESCRIPTION
**Pull request by MarcAbonce**
## What does this PR do?

Some of our interface locales include uppercase country codes, which are separated by `_` instead of the more common `-`.
Also, a browser's `Accept-Language` header could be in lowercase.

This commit attempts to normalize those cases so a browser's language+country codes can better match with our locales.

## How to test this PR locally?

1. Change your browser's `Accept-Language` header to add a country specific locale that we support (such as `pt-BR`, `nl-BE` or `zh-TW`). Note: This locale has to be the first one in the `Accept-Language` list.
2. Go to preferences.

Expected result: Correct locale is selected (`pt_BR` rather than `pt` and so on).

I also added a unit test to automate this.

## Author's checklist
This solution assumes that our UI locales have nothing more than language and optionally a country.
If we ever add a script specific locale like `zh_Hant_TW`* this would have to change to accommodate that, but the idea would be pretty much the same as this fix.

*By the way, in [our Transifex page](https://www.transifex.com/asciimoo/searx/languages/) it looks like `zh_Hant_TW` was removed (I assume it was merged with `zh_TW` which is already in traditional Chinese). Should we remove [this directory](https://github.com/searx/searx/tree/master/searx/translations/zh_Hant_TW/LC_MESSAGES) or am I making a wrong assumption here?

## Related issues
Closes #2525

### CODE REVIEW

**Review by return42 — DISMISSED** [review]
I made a slightly different test:

1. In the settings I added `de_AT : Deutsch (Austria)` to the _locales_ section.
2. In my browser I set `Accept-Language: de-at`

> Expected result: Correct locale is selected (pt_BR rather than pt and so on).

I tested this on branch `master` and on your `MarcAbonce:fix-browser-locale`.  Both with the same result; in the preferences the search-language is 'de-AT'.

With your patch applied, the only difference I see is:

branch `master` :

    DEBUG:searx.webapp:http://127.0.0.1:8888/preferences uses locale `de` from browser
    DEBUG:searx.webapp:http://127.0.0.1:8888/preferences uses locale `de` from preferences

branch `MarcAbonce:fix-browser-locale`

    DEBUG:searx.webapp:http://127.0.0.1:8888/ uses locale `de_AT` from browser
    DEBUG:searx.webapp:http://127.0.0.1:8888/ uses locale `de_AT` from preferences

To summarize: from my POV this patch fixes the issue, but I needed to check the debug log.

**Review by laichiaheng — DISMISSED** [review]

**Comment by kvch** [comment]
@MarcAbonce Please update the PR, so I can merge it.

**Review by kvch — DISMISSED** [review]

**Review by laichiaheng — APPROVED** [review]

**Review by dalf — APPROVED** [review]
Thank you!
